src/Documents/Document.ts

```diff
@@ -0,0 +1,15 @@
+import {IDocument} from './IDocument';
+
+export class Document implements IDocument {
+  public entityName(): string {
+      return this.constructor.name;
+  }
+
```

Comment:
`JSON.serialize` should handle it, no?
And validation is something that you do outside of the object, using listeners, for example.